lib/execution_engine2/db/MongoUtil.py

```diff
@@ -574,80 +478,15 @@ def _push_job_logs(self, log_lines: JobLog, job_id: str, record_count: int):
-            except Exception as e:
-                error_msg = "Cannot update doc\n ERROR -- {}:\n{}".format(
-                    e, "".join(traceback.format_exception(None, e, e.__traceback__))
-                )
-                raise ValueError(error_msg)
-
-            slc = job_col.find_one({"_id": ObjectId(job_id)}).get("stored_line_count")
-
-            return slc
-
-    def update_one(self, doc, job_id):
-        """
-        update existing records or create if they do not exist
-        https://docs.mongodb.com/manual/reference/operator/update/set/
-        """
-        with self.pymongo_client(self.mongo_collection) as pymongo_client:
-            job_col = pymongo_client[self.mongo_database][self.mongo_collection]
-            try:
-                update_filter = {"_id": ObjectId(job_id)}
-                update = {"$set": doc}
-                job_col.update_one(update_filter, update, upsert=True)
-            except Exception as e:
-                error_msg = "Connot update doc\n"
-                error_msg += "ERROR -- {}:\n{}".format(
-                    e, "".join(traceback.format_exception(None, e, e.__traceback__))
-                )
-                raise ValueError(error_msg)
-
-        return True
-
-    def delete_one(self, job_id):
-        """
-        delete a doc by _id
-        """
-        self.logger.debug("start deleting document")
-        with self.pymongo_client(self.mongo_collection) as pymongo_client:
-            job_col = pymongo_client[self.mongo_database][self.mongo_collection]
-            try:
-                delete_filter = {"_id": ObjectId(job_id)}
-                job_col.delete_one(delete_filter)
-            except Exception as e:
-                error_msg = "Connot delete doc\n"
-                error_msg += "ERROR -- {}:\n{}".format(
-                    e, "".join(traceback.format_exception(None, e, e.__traceback__))
-                )
-                raise ValueError(error_msg)
-
-        return True
-
-    def find_in(self, elements, field_name, projection=None, batch_size=1000):
-        """
-        return cursor that contains docs which field column is in elements
-        """
-        self.logger.debug("start querying MongoDB")
-
-        with self.pymongo_client(self.mongo_collection) as pymongo_client:
-            job_col = pymongo_client[self.mongo_database][self.mongo_collection]
-            try:
-                result = job_col.find(
-                    {field_name: {"$in": elements}},
-                    projection=projection,
-                    batch_size=batch_size,
-                )
-            except Exception as e:
-                error_msg = "Connot query doc\n"
-                error_msg += "ERROR -- {}:\n{}".format(
-                    e, "".join(traceback.format_exception(None, e, e.__traceback__))
-                )
-                raise ValueError(error_msg)
+        job_col = self.pymongoc[self.mongo_database][self._col_logs]
```

Comment:
This method, if run when the old `mongo_collection` variable was set to the jobs collection, could add log information to the job record

Reply:
I also assume it'd fail with some sort of null pointer-ish error if it was called before one of the other methods that set `mongo_collection` had been called.